### ISSUE TITLE
Can you provide support to get the property names

### ISSUE THREAD

**Issue by ceepan** [json-path/JsonPath#417]
Looking at https://github.com/s3u/JSONPath it allow the use of the tilde character to get all of the property names.  Example is taking the following JSON:

{
    "store": {
      "book": [
        {
          "category": "reference",
          "author": "Nigel Rees",
          "title": "Sayings of the Century",
          "price": 8.95
        },
        {
          "category": "fiction",
          "author": "Evelyn Waugh",
          "title": "Sword of Honour",
          "price": 12.99
        },
        {
          "category": "fiction",
          "author": "Herman Melville",
          "title": "Moby Dick",
          "isbn": "0-553-21311-3",
          "price": 8.99
        },
        {
          "category": "fiction",
          "author": "J. R. R. Tolkien",
          "title": "The Lord of the Rings",
          "isbn": "0-395-19395-8",
          "price": 22.99
        }
      ],
      "bicycle": {
        "color": "red",
        "price": 19.95
      }
    }
  }

and using the JSONPath expression "$.store.*~" to return "["book","bicycle"]"

The implementation behind http://jsonpath.com/ supports this functionality

**Comment by pavelkostyurin** [comment]
Any hope of implementing this feature?

**Comment by miparnisari** [comment]
Ping :) any hope?

**Comment by Shirinrpatel** [comment]
Hey guys, can someone please look into this? thanks

**Comment by fghber** [comment]
> Looks like a duplicate of #439

Agreed. Arguably, this is a generalization of #439 i.e. get the property name anywhere.
